src/nvim/tui/input.c

```diff
@@ -376,8 +412,8 @@ static void read_cb(Stream *stream, RBuffer *buf, size_t c, void *data,
     }
   } while (rbuffer_size(input->read_stream.buffer));
   flush_input(input, true);
-  // Make sure the next input escape sequence fits into the ring buffer
-  // without wrap around, otherwise it could be misinterpreted.
+  // CSI and K_SPECIAL are double-escaped later (input_enqueue).
+  // Make sure they can fit into the ring buffer without wrap around.
   rbuffer_reset(input->read_stream.buffer);
```

Comment:
No, `input->read_stream.buffer` is only read by the loop above. There could be an multi-byte key sequence that is trimmed by the end of the buffer(wrapping around to the beginning), but `termkey_push_bytes` is not aware of that, so it could misinterpret the key. `rbuffer_reset` simply moves the data so it is all in contiguous memory.